diff --git a/dist/chunk-G2K6T2QU.mjs b/dist/chunk-G2K6T2QU.mjs
index f218547a8295a85dfd6ff077da88897d369eb5d0..2220c477ebac6d8307b6f871772b677baa1c646f 100644
--- a/dist/chunk-G2K6T2QU.mjs
+++ b/dist/chunk-G2K6T2QU.mjs
@@ -69,7 +69,7 @@ function useChat(options) {
     abortControllerRef.current = abortController;
     setMessages([...previousMessages, ...newMessages]);
     const systemMessage = makeSystemMessageCallback();
-    const messagesWithContext = [systemMessage, ...initialMessages || [], ...previousMessages];
+    const messagesWithContext = [systemMessage, ...initialMessages || [], ...previousMessages.slice(-12)];
     const stream = CopilotRuntimeClient.asStream(
       runtimeClient.generateCopilotResponse({
         data: __spreadProps(__spreadValues(__spreadProps(__spreadValues({
diff --git a/dist/hooks/use-chat.js b/dist/hooks/use-chat.js
index 1b8d088bb2691c2222e56cf0fb9bb67d7663159e..dca6eb4b9288c08e97e373953f0c2fc55d15e2b6 100644
--- a/dist/hooks/use-chat.js
+++ b/dist/hooks/use-chat.js
@@ -109,7 +109,7 @@ function useChat(options) {
     abortControllerRef.current = abortController;
     setMessages([...previousMessages, ...newMessages]);
     const systemMessage = makeSystemMessageCallback();
-    const messagesWithContext = [systemMessage, ...initialMessages || [], ...previousMessages];
+    const messagesWithContext = [systemMessage, ...initialMessages || [], ...previousMessages.slice(-12)];
     const stream = import_runtime_client_gql.CopilotRuntimeClient.asStream(
       runtimeClient.generateCopilotResponse({
         data: __spreadProps(__spreadValues(__spreadProps(__spreadValues({
